dnsdist: Warning when building the BPF filter code on CentOS 8

- Program: dnsdist <!-- delete the ones that do not apply -->
 - Issue type: Bug report

### Short description
I just saw these warnings in the CentOS 8 build log:
```
bpf-filter.cc:50:5: warning: no previous declaration for 'int bpf_pin_map(int, const string&)' [-Wmissing-declarations]
 int bpf_pin_map(int fd, const std::string& path)
     ^~~~~~~~~~~
bpf-filter.cc:59:5: warning: no previous declaration for 'int bpf_load_pinned_map(const string&)' [-Wmissing-declarations]
 int bpf_load_pinned_map(const std::string& path)
     ^~~~~~~~~~~~~~~~~~~
bpf-filter.cc:67:6: warning: no previous declaration for 'void bpf_check_map_sizes(int, uint32_t, uint32_t)' [-Wmissing-declarations]
 void bpf_check_map_sizes(int fd, uint32_t expectedKeySize, uint32_t expectedValueSize)
      ^~~~~~~~~~~~~~~~~~~
bpf-filter.cc: In member function 'void BPFFilter::block(const DNSName&, BPFFilter::MatchAction, uint16_t)':
bpf-filter.cc:478:42: warning: 'void* memset(void*, int, size_t)' clearing an object of non-trivial type 'struct CounterAndActionValue'; use assignment or value-initialization instead [-Wclass-memaccess]
     memset(&cadvalue, 0, sizeof(cadvalue));
                                          ^
bpf-filter.cc:192:8: note: 'struct CounterAndActionValue' declared here
 struct CounterAndActionValue
        ^~~~~~~~~~~~~~~~~~~~~
bpf-filter.cc:484:38: warning: 'void* memset(void*, int, size_t)' clearing an object of non-trivial type 'struct QNameValue'; use assignment or value-initialization instead [-Wclass-memaccess]
     memset(&qvalue, 0, sizeof(qvalue));
                                      ^
bpf-filter.cc:186:8: note: 'struct QNameValue' declared here
 struct QNameValue
        ^~~~~~~~~~
```

I will have to fix these before 1.7.0 final.

Also a warning reported by clang++:
```
remote_logger.cc:50:10: warning: variable 'total' set but not used [-Wunused-but-set-variable]
  size_t total = 0;
```

--- a/pdns/bpf-filter.cc
+++ b/pdns/bpf-filter.cc
@@ -35,19 +35,8 @@ static __u64 ptr_to_u64(void *ptr)
   return (__u64) (unsigned long) ptr;
 }
 
-int bpf_create_map(enum bpf_map_type map_type, int key_size, int value_size,
-                   int max_entries)
-{
-  union bpf_attr attr;
-  memset(&attr, 0, sizeof(attr));
-  attr.map_type = map_type;
-  attr.key_size = key_size;
-  attr.value_size = value_size;
-  attr.max_entries = max_entries;
-  return syscall(SYS_bpf, BPF_MAP_CREATE, &attr, sizeof(attr));
-}
-
-int bpf_pin_map(int fd, const std::string& path)
+/* these can be static as they are not declared in libbpf.h: */
+static int bpf_pin_map(int fd, const std::string& path)
 {
   union bpf_attr attr;
   memset(&attr, 0, sizeof(attr));
@@ -56,7 +45,7 @@ int bpf_pin_map(int fd, const std::string& path)
   return syscall(SYS_bpf, BPF_OBJ_PIN, &attr, sizeof(attr));
 }
 
-int bpf_load_pinned_map(const std::string& path)
+static int bpf_load_pinned_map(const std::string& path)
 {
   union bpf_attr attr;
   memset(&attr, 0, sizeof(attr));
@@ -64,7 +53,7 @@ int bpf_load_pinned_map(const std::string& path)
   return syscall(SYS_bpf, BPF_OBJ_GET, &attr, sizeof(attr));
 }
 
-void bpf_check_map_sizes(int fd, uint32_t expectedKeySize, uint32_t expectedValueSize)
+static void bpf_check_map_sizes(int fd, uint32_t expectedKeySize, uint32_t expectedValueSize)
 {
   struct bpf_map_info info;
   uint32_t info_len = sizeof(info);
@@ -89,6 +78,18 @@ void bpf_check_map_sizes(int fd, uint32_t expectedKeySize, uint32_t expectedValu
   if (info.value_size != expectedValueSize) {
     throw std::runtime_error("Error checking the size of eBPF map: value size mismatch (" + std::to_string(info.value_size) + " VS " + std::to_string(expectedValueSize) + ")");
   }
+}
+
+int bpf_create_map(enum bpf_map_type map_type, int key_size, int value_size,
+                   int max_entries)
+{
+  union bpf_attr attr;
+  memset(&attr, 0, sizeof(attr));
+  attr.map_type = map_type;
+  attr.key_size = key_size;
+  attr.value_size = value_size;
+  attr.max_entries = max_entries;
+  return syscall(SYS_bpf, BPF_MAP_CREATE, &attr, sizeof(attr));
 }
 
 int bpf_update_elem(int fd, void *key, void *value, unsigned long long flags)
@@ -475,13 +476,11 @@ void BPFFilter::block(const DNSName& qname, BPFFilter::MatchAction action, uint1
   void* value = nullptr;
 
   if (d_external) {
-    memset(&cadvalue, 0, sizeof(cadvalue));
     cadvalue.counter = 0;
     cadvalue.action = action;
     value = &cadvalue;
   }
   else {
-    memset(&qvalue, 0, sizeof(qvalue));
     qvalue.counter = 0;
     qvalue.qtype = qtype;
     value = &qvalue;

--- a/pdns/dnsdistdist/dnsdist-tcp-downstream.cc
+++ b/pdns/dnsdistdist/dnsdist-tcp-downstream.cc
@@ -515,7 +515,9 @@ void TCPConnectionToBackend::notifyAllQueriesFailed(const struct timeval& now, F
   /* we might be terminated while notifying a query sender */
   d_ds->outstanding -= d_pendingResponses.size();
   auto pendingQueries = std::move(d_pendingQueries);
+  d_pendingQueries.clear();
   auto pendingResponses = std::move(d_pendingResponses);
+  d_pendingResponses.clear();
 
   auto increaseCounters = [reason](std::shared_ptr<TCPQuerySender>& sender) {
     if (reason == FailureReason::timeout) {

--- a/pdns/remote_logger.cc
+++ b/pdns/remote_logger.cc
@@ -47,12 +47,10 @@ bool CircularWriteBuffer::flush(int fd)
 
   struct iovec iov[2];
   int pos = 0;
-  size_t total = 0;
   for(const auto& arr : {arr1, arr2}) {
     if(arr.second) {
       iov[pos].iov_base = arr.first;
       iov[pos].iov_len = arr.second;
-      total += arr.second;
       ++pos;
     }
   }
@@ -84,7 +82,6 @@ bool CircularWriteBuffer::flush(int fd)
   }
   while (res < 0);
 
-  //  cout<<"Flushed "<<res<<" bytes out of " << total <<endl;
   if (static_cast<size_t>(res) == d_buffer.size()) {
     d_buffer.clear();
   }

